test/test_init_version.py

```diff
@@ -0,0 +1,63 @@
+import torch
+import torch.nn as nn
+from torch.nn.init import init_version, _init_version, _parse_version
+
+from torch.testing._internal.common_utils import TestCase, run_tests
+
+class InitVersionTest(TestCase):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+    def test_pre_1_7_0_linear(self):
+        # uniform init is used
+        torch.manual_seed(4)
+        with init_version('1.6.1') as v:
+            l = nn.Linear(2, 3)
+
+        x = l.bias.mean().item()
+        y = 0.48299
+        self.assertEqual(x, y)
+
+    def test_1_7_0_linear(self):
+        # zero bias is used
+        with init_version('1.7.0') as v:
+            l = nn.Linear(2, 3)
+
+        x = l.bias.data
+        y = torch.zeros(3, dtype=torch.float32)
+        self.assertEqual(x, y)
+
+    def test_init_version_remains_same(self):
+        # _init_version should remain same before and after using init_version
+        before = _init_version
+        with init_version() as v:
+            l = nn.Linear(2, 3)
+        after = _init_version
+
+        x = torch.tensor(before)
+        y = torch.tensor(after)
+        self.assertEqual(x, y)
+
+    def test_version_greater_than_torch(self):
+        # cannot pass version greater than torch.__version__ in init_version
```

Comment:
If current_torch_version = (1,7,0), then I pass (1,8,0) to `use_init_version`. Now it goes through these checks
1. `use_init_version` raise error if version greater
2. If the above does not raise error, initialize a linear layer to see if it raises an error
3. If the above does not raise error, then raise error that something is wrong.

In the `except` clause I match the error message so that it only checks for the "version greater than" one.